src/sync.rs

```diff
@@ -140,11 +146,13 @@ struct Futex {
 
 /// A thread waiting on a futex.
 #[derive(Debug)]
-struct FutexWaiter {
+pub struct FutexWaiter {
     /// The thread that is waiting on this futex.
-    thread: ThreadId,
+    pub thread: ThreadId,
     /// The bitset used by FUTEX_*_BITSET, or u32::MAX for other operations.
-    bitset: u32,
+    pub bitset: u32,
+    /// The value used by WaitOnAddress & WakeByAddressSingle to determine if a thread should wake
```

Comment:
This file should be comprehensible to people that are not familiar with windows. So please extend this comment since I have no idea what you mean by this.

Also do we have to make this type public...?